opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -384,16 +384,15 @@ namespace Opm {
         if (!wtest_config.empty()) { // there is a WTEST request
             const std::vector<std::string> wellsForTesting = wellTestState()
                 .test_wells(wtest_config, simulationTime);
-
             for (const std::string& well_name : wellsForTesting) {
-                const auto& ws = this->wellState().well(well_name);
-                if (ws.status != Well::Status::OPEN)
-                    continue;
 
                 WellInterfacePtr well = createWellForWellTest(well_name, timeStepIdx, deferred_logger);
                 // some preparation before the well can be used
                 well->init(&phase_usage_, depth_, gravity_, local_num_cells_, B_avg_);
                 const Well& wellEcl = schedule().getWell(well_name, timeStepIdx);
+                if (wellEcl.getStatus() == Well::Status::SHUT)
+                    continue;
+
```

Comment:
Two comments/questions:

1. I guess the check could be moved to the top of the body of the loop? 
2. I guess the check could be `wellEcl.getStatus() != Well::Status::OPEN)`

Reply:
1) done. 2) I think stopped wells should be tested